### PR TITLE
Move to error prone

### LINKED ISSUE #1141: investigate migrating from findbugs to errorprone

### PR DESCRIPTION
**Pull request by patflynn**
preliminary. PTAL.  I won't check this in until after the release.  Also need to add instructions in the PR for setup IJ compiler.

fixes #1141 

Benefits are:
Less noise.
No config.
You never will push a PR and find that Jenkins has found an error, because those errors will fail your build and also show up as errors in your IJ editor.

@aslo @etanshaul @joaoandremartins

### CODE REVIEW

**Comment by chanseokoh** [comment]
Also need to remove the `@SuppressFBWarnings` annotation. There are a couple of instances.

**Comment by patflynn** [comment]
Good point. also removed the dependency in build.gradle.

File: .idea/compiler.xml [hunk 1]
```diff
@@ -27,10 +27,6 @@
       <module name="google-account-plugin_test" target="1.7" />
       <module name="google-cloud-tools-plugin_main" target="1.7" />
       <module name="google-cloud-tools-plugin_test" target="1.7" />
-      <module name="jps-plugin_main" target="1.7" />
```
Comment (joaoandremartins) [inline .idea/compiler.xml]: After the source context change, we'll need to leave this bit here I think (even though I'm not too sure what this does).

File: .idea/google-java-format.xml [hunk 2]
```diff
@@ -2,5 +2,45 @@
...
+        <entry key="common-test-lib_main" value="ENABLED" />
+        <entry key="common-test-lib_test" value="ENABLED" />
+        <entry key="gcloud-intellij" value="ENABLED" />
+        <entry key="google-account-plugin" value="ENABLED" />
+        <entry key="google-account-plugin_main" value="ENABLED" />
+        <entry key="google-account-plugin_test" value="ENABLED" />
+        <entry key="google-cloud-tools-plugin" value="ENABLED" />
+        <entry key="google-cloud-tools-plugin_main" value="ENABLED" />
+        <entry key="google-cloud-tools-plugin_test" value="ENABLED" />
+        <entry key="ultimate" value="ENABLED" />
```
Comment (joaoandremartins) [inline .idea/google-java-format.xml]: Mental note to also add jps-plugin here after the source context merge.

**Comment by patflynn** [comment]
I'll wait until you merge your change first, and then rebase.

File: gradle.properties [hunk 3]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-ideaEdition = IC
+ideaEdition = IU
```
Comment (joaoandremartins) [inline gradle.properties]: Is this intended?

**Comment by patflynn** [comment]
nope.

File: gradle.properties (hunk 3 above)
Comment (aslo) [inline gradle.properties]: @patflynn Why not keep this? IU is our main use case anyway. I personally always manually change this to IU when running locally.

**Comment by patflynn** [comment]
Mainly it's for the CI environment.

One convenient way I've found of handling this is adding the file to a
separate change list in the IntelliJ VCS panel. It then typically
intelligently stashes and reapplies as I switch between branches, and it's
not included in pushes (except accidentally in this case).

**Comment by patflynn** [comment]
@aslo @etanshaul @joaoandremartins OK I think this is ready to merge.  PTAL and follow the CONTRIBUTING instructions after I merge.

File: .idea/compiler.xml [hunk 4]
```diff
@@ -13,11 +13,6 @@
       <entry name="!?*.kt" />
       <entry name="!?*.clj" />
     </wildcardResourcePatterns>
-    <annotationProcessing>
```
Comment (joaoandremartins) [inline .idea/compiler.xml]: Why is this removed?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ui/CloudDebugHistoricalSnapshots.java [hunk 5]
```diff
@@ -648,7 +648,7 @@ public void run() {
       for (int row = 0; row < getModel().getRowCount(); row++) {
         // todo: getModel should be newModel
         Breakpoint bp = getModel().getBreakpoints().get(row);
-        if (bp.getIsFinalState() != Boolean.TRUE) {
+        if (bp.getIsFinalState().equals(Boolean.TRUE)) {
```
Comment (joaoandremartins) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ui/CloudDebugHistoricalSnapshots.java]: I think you need a "!" before bp?

File: gradle.properties [hunk 6]
```diff
@@ -13,8 +13,7 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-
-ideaEdition = IC
+ideaEdition=IC
```
Comment (joaoandremartins) [inline gradle.properties]: Maybe unstage gradle.properties?

File: google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ui/CloudDebugHistoricalSnapshots.java (hunk 5 above)
Comment (patflynn) [inline google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/debugger/ui/CloudDebugHistoricalSnapshots.java]: well shit! yeah. Good catch!

File: .idea/compiler.xml (hunk 4 above)
Comment (patflynn) [inline .idea/compiler.xml]: I didn't actually modify this file. It was the IJ settings editor that updated this when I switched the compiler.  Looking at our old and new settings UI (before and after the switch), it doesn't appear that annotation processors were ever on.

**Comment by patflynn** [comment]
@joaoandremartins fixed. GoogleJavaFormat already doing it's magic.  I didn't realize it would rearrange method declarations.

**Comment by patflynn** [comment]
any other comments?

File: gradle.properties [hunk 7]
```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 #
 
-ideaEdition = IC
+ideaEdition=IC
```
Comment (joaoandremartins) [inline gradle.properties]: nit: probably add spaces before and after the equals sign

**Comment by patflynn** [comment]
Aha I had! But it turns out my IJ formatter is somehow messing this up.
fixed.
